kolibri/core/assets/src/api-resource.js

```diff
@@ -186,7 +186,7 @@ class Model {
             this.resource.removeModel(this);
```

Comment:
This section doesn't actually make sense.

If the model doesn't have an id, then it won't be in the model cache.

Also, then resolving it with `this.id` which we know is not defined inside this `if` block also makes no sense.

Might be better to raise an error here saying "Can't delete model that we don't have an id for".